descartes_planner/src/sparse_planner.cpp

```diff
@@ -589,9 +601,26 @@ void SparsePlanner::sampleTrajectory(double sampling,const std::vector<Trajector
   int skip = std::ceil(double(1.0f)/sampling);
   ROS_INFO_STREAM("Sampling skip val: "<<skip<< " from sampling val: "<<sampling);
   ss<<"[";
-  for(int i = 0; i < dense_trajectory_array.size();i+=skip)
-  {
-    sparse_trajectory_array.push_back(dense_trajectory_array[i]);
+  
+  if (dense_trajectory_array.empty()) return;
+
+  // Add the first point
+  sparse_trajectory_array.push_back(dense_trajectory_array.front());
+  ss << "0 ";
+  // The first point requires no special timing adjustment
+  
+  int i; // We keep i outside of the loop so we can examine it on the last step
+  for (i = skip; i < dense_trajectory_array.size(); i+=skip)
```

Comment:
Skip is defined on line 601 of the new file to be:
```c++
int skip = std::ceil(double(1.0f)/sampling);
```

This is unchanged from the original implementation.